packages/integration/src/azure/config.ts

```diff
@@ -33,25 +33,26 @@ export type AzureIntegrationConfig = {
   host: string;
 
   /**
-   * The authorization token to use for requests.
+   * The credentials to use for requests. If multiple credentials are specified the first one that matches the organization is used.
+   * If not organization matches the first credential without an organization is used.
    *
-   * If no token is specified, anonymous access is used.
+   * If no credentials are specified at all, anonymous access is used.
    */
-  token?: string;
+  credentials?: AzureIntegrationCredentialConfig[];
+};
 
+type AzureIntegrationCredentialConfigBase = {
   /**
-   * The credential to use for requests.
-   *
-   * If no credential is specified anonymous access is used.
+   * The Azure DevOps organizations for which to use this credential.
    */
-  credential?: AzureCredential;
+  organizations?: string[];
 };
 
 /**
  * Authenticate using a client secret that was generated for an App Registration.
  * @public
  */
-export type ClientSecret = {
+export type ClientSecretCredential = AzureIntegrationCredentialConfigBase & {
```

Comment:
I like this rename and I think we could take it one step further, `AzureClientSecretCredential`. After the release next week this will be a breaking change. What do you think about moving these renames to a separate PR that we can ship a bit faster?

Reply:
Sure thing, see #18273. I'll rebase this branch on those changes